source/extensions/dynamic_modules/sdk/rust/src/lib.rs

```diff
@@ -171,9 +270,110 @@ unsafe extern "C" fn envoy_dynamic_module_on_http_filter_config_destroy(
   let _inner = Box::from_raw(*config);
 }
 
+#[no_mangle]
+unsafe extern "C" fn envoy_dynamic_module_on_http_filter_new(
+  filter_config_ptr: abi::envoy_dynamic_module_type_http_filter_config_module_ptr,
+  filter_envoy_ptr: abi::envoy_dynamic_module_type_http_filter_envoy_ptr,
+) -> abi::envoy_dynamic_module_type_http_filter_module_ptr {
+  let envoy_filter_config = EnvoyHttpFilterConfig {
+    raw_ptr: filter_envoy_ptr,
+  };
+  let filter_config = {
+    let raw = filter_config_ptr as *mut *mut dyn HttpFilterConfig;
+    &**raw
+  };
+  envoy_dynamic_module_on_http_filter_new_impl(envoy_filter_config, filter_config)
+}
+
+fn envoy_dynamic_module_on_http_filter_new_impl(
+  envoy_filter_config: EnvoyHttpFilterConfig,
+  filter_config: &dyn HttpFilterConfig,
+) -> abi::envoy_dynamic_module_type_http_filter_module_ptr {
+  let filter = filter_config.new_http_filter(envoy_filter_config);
+  let boxed = Box::into_raw(Box::new(filter));
+  boxed as abi::envoy_dynamic_module_type_http_filter_module_ptr
+}
+
+#[no_mangle]
+unsafe extern "C" fn envoy_dynamic_module_on_http_filter_destroy(
+  filter_ptr: abi::envoy_dynamic_module_type_http_filter_module_ptr,
+) {
+  let filter = filter_ptr as *mut *mut dyn HttpFilter;
+
+  // Drop the Box<*mut dyn HttpFilter>, and then the Box<dyn HttpFilter>, which also drops the
+  // underlying object.
+  let _outer = Box::from_raw(filter);
+  let _inner = Box::from_raw(*filter);
```

Comment:
Sorry what is this about?

Reply:
yeah so this is tricky thing and already merged in the previous PR but looks like due to some drift I accidentally deleted the code comments.

Done in https://github.com/envoyproxy/envoy/pull/37325/commits/2fa78ca8170095b7299eabdde20d8ec11e0110fa for the future reference by centralizing the logic and I added an explicit explanation. tldr is that we are using raw pointers as a communication, and on the Rust SDK implementation-side, we need to retrieve the type information (e.g. which Trait implementation is currently called on event hooks) to call the right callbacks.